src/metabase/feature_extraction/automagic_dashboards.clj

```diff
@@ -0,0 +1,350 @@
+(ns metabase.feature-extraction.automagic-dashboards
```

Comment:
this namespace should be broken up into a couple. I foresee layout being its own beast, and constraint matching is already complex enough that it should probably be in its own namespace.